WordPress/WordPressTest/TestUtilities/JSONObject.swift

```diff
@@ -21,20 +22,11 @@ extension JSONObject {
     ///   - name: The name of the file
     ///   - type: The extension of the file
     /// - Returns: A dictionary representing the contents of the JSON file.
-    static func loadFile(_ name: String, type: String) -> JSONObject {
-        guard let url = Bundle(for: BundlerFinder.self).url(forResource: name, withExtension: type) else {
-            fatalError("File not found in the test bundle: \(name).\(type)")
-        }
-        guard let data = try? Data(contentsOf: url) else {
-            fatalError("Can't read content of file: \(name).\(type)")
-        }
-        guard let parseResult = try? JSONSerialization.jsonObject(with: data, options: [.mutableContainers, .mutableLeaves]) else {
-            fatalError("Can't parse file as JSON: \(name).\(type)")
-        }
-        guard let result = parseResult as? JSONObject else {
-            fatalError("File content isn't a JSON object: \(name).\(type)")
-        }
-        return result
+    static func loadFile(_ name: String, type: String) throws -> JSONObject {
```

Comment:
Totally out of scope suggestion: It just occurred to me that we could simplify some of this logic by implicitly expecting the type (extension) to be JSON.

After all this is part of `JSONObject.swift`!

We could have something like 'if input is "file-name", assume extension to be JSON; if input "file-name.extension", verify extension is JSON and throw an error otherwise.'

What do you think? Again, this is out of scope for this PR and only a minor improvement.

Reply:
Good call! Considering `JSONObject` is just a specialised `Dictionary`, which means `JSONObject.loadFile(named: "foo.json")` is equivalent to `Dictionary<String, AnyObject>.loadFile(named:"foo.json")`. Do you think, on top of your proposal, is it worth making the function name more explicit, like changing it to `loadJSONFile(named:)`?

Reply:
> Do you think, on top of your proposal, is it worth making the function name more explicit, like changing it to `loadJSONFile(named:)`?

Good point! Also because most of the usages are in one liner methods where `.loadJSONFile(named:)` is clearer than `.loadFile(named:)`.

👍